server/src/lsp-server.ts

```diff
@@ -75,16 +75,21 @@ export class LspServer {
         if (this.options.tsserverPath) {
             return this.options.tsserverPath;
         }
-        // 1) look into node_modules of workspace root
+        // 1) look into .yarn/sdks of workspace root
+        const sdk = findPathToSdk(this.rootPath(), path.join('typescript', 'lib', 'tsserver.js'));
+        if (sdk) {
+            return sdk;
+        }
+        // 2) look into node_modules of workspace root
         const executable = findPathToModule(this.rootPath(), `.bin/${getTsserverExecutable()}`);
         if (executable) {
             return executable;
         }
-        // 2) use globally installed tsserver
+        // 3) use globally installed tsserver
```

Comment:
I'm thinking about a case where:
 - the current project uses npm (so no any PNP magic)
 - The user has yarn sdk initialized at `~/.yarn/sdk/...` (is that even an option?)

Assuming that that's a possible scenario, what would be an expected behavior here? I think picking yarn sdk version would be incorrect.

Reply:
Yeah I think so but as I understand it yarn v2 (and PnP) is setup per project so I don't think it's possible to have a user level setup for sdks. @merceyz do you know if that is correct?

Reply:
Correct, there should be no need to check the home directory for the SDK

Reply:
Thanks for clarifying.

Just to clarify a bit more, the current logic will check for `.yarn/sdk/...` up the directory tree, eventually ending up checking the user directory also (assuming that the project lives within the user's directory somewhere, which is likely the case).

So I'm wondering if we should short-cut that check to stop the traverse early.

Should it for example only traverse up to the first `package.json` that it finds?
Or should it in fact not traverse at all and just check within the `rootPath()`)?

Reply:
I'm not too sure how you'd concisely express the "correct" algorithm. It _should_ traverse up until it leaves the repo, checking every site it finds a `package.json` file. Not finding a `.yarn/sdks` alongside a `package.json` isn't necessarily grounds to terminate as it could be a workspace configuration in which case there is another `package.json` higher up which could have the `.yarn/sdks` folder 😬

Reply:
OK. The current behavior is probably fine then.